modules/ROOT/pages/type-definitions/directives/custom-logic.adoc

```diff
@@ -0,0 +1,641 @@
+            RETURN a
+            """,
+            columnName: "a"
+        )
+}
+----
+
+
+== `@coalesce`
+
+When translating from GraphQL to Cypher, any instances of fields to which this directive is applied will be wrapped in a `coalesce()` function in the WHERE clause (see https://neo4j.com/developer/kb/understanding-non-existent-properties-and-null-values/#_use_coalesce_to_use_a_default_for_a_null_value).
+
+This directive helps querying against non-existent properties in a database. 
+However, it is encouraged to populate these properties with meaningful values if this is becoming the norm. The `@coalesce`directive is a primitive implementation of the function which only takes a static default value as opposed to using another property in a node or a Cypher expression.
+
+=== Definition
+
+[source, graphql, indent=0]
+----
+"""Int | Float | String | Boolean | ID | DateTime | Enum"""
+scalar ScalarOrEnum
+
+"""Instructs @neo4j/graphql to wrap the property in a coalesce() function during queries, using the single value specified."""
+directive @coalesce(
+    """The value to use in the coalesce() function. Must be a scalar type and must match the type of the field with which this directive decorates."""
+    value: Scalar!,
+) on FIELD_DEFINITION
+----
+
+=== Usage
+
+`@coalesce` may be used with enums. 
+When setting the default value for an enum field, it must be one of the enumerated enum values:
+
+[source, graphql, indent=0]
+----
+enum Status {
+    ACTIVE
+    INACTIVE
+}
+type Movie {
+    status: Status @coalesce(value: ACTIVE)
+}
+----
+
+[[type-definitions-default-values-limit]]
+== `@limit`
+
+Available on nodes, this directive injects values into a query such as the `limit`.
+
+=== Definition
+
+[source, graphql, indent=0]
+----
+"""The `@limit` is to be used on nodes, where applied will inject values into a query such as the `limit`."""
+directive @limit(
+    default: Int
+    max: Int
+) on OBJECT
+----
+
+=== Usage
+
+The directive has two arguments:
+
+* `default` - if no `limit` argument is passed to the query, the default limit is used. 
+The query may still pass a higher or lower `limit`.
+* `max` - defines the maximum limit to be passed to the query. 
+If a higher value is passed, it is used instead. 
+If no `default` value is set, `max` is used for queries without limit.
+
+[source, graphql, indent=0]
+----
+{
+  Movie @limit(amount: 5) {
+    title
+    year
+  }
+}
+----
+
+== `@customResolver`
+
+The Neo4j GraphQL Library generates query and mutation resolvers, so you don't need to implement them yourself.
+However, if you need additional behaviors besides the autogenerated CRUD operations, you can specify custom resolvers for these scenarios.
+This page describes how to set them up.
+
+To add a field to an object type which is resolved from existing values in the type, rather than storing new values, you should mark it with the `@customResolver` directive and define a custom resolver for it.
```

Comment:
```suggestion
To add a field to an object type which is resolved from existing values in the type, rather than storing new values, you should mark it with the `@customResolver` directive, and define a custom resolver for it.
```